ios/Trifle/Sources/Trifle.swift

```diff
@@ -117,53 +118,64 @@ public class Trifle {
             certificates: certificates
         )
     }
-}
-
-extension Certificate {
+    
     /**
-     Verify that the provided certificate matches what we expected.
-     It matches the CSR that we have and the root cert is what
-     we expect.
-
-     - parameters: certificateRequest request used to generate this certificate
-     - parameters: certificateChain - list of certificates between this cert and
-        the root certificate.
-     - parameters: rootCertificate - certificate to use as root of chain.
-        Defaults to the root certificate bundled with Trifle.
+     Sign the provided data with the provided key, including appropriate Trifle
+     metadata, such as the accompanying Trifle certificate.
      
-     - returns: true if validated, false otherwise
+     - parameters: data - raw data to be signed.
+     - parameters: keyHandle - key handle used for the signing.
+     - parameters: certificate list - Trifle certificate chain to be included in the SignedData message.
+     Must match the key in keyHandle.
+     
+     - returns:`SignedData` - signed data message in the Trifle format.
      */
-    public func verify(
-        certificateRequest: MobileCertificateRequest? = nil,
-        intermediateChain: Array<Certificate>,
-        rootCertificate: Certificate? = nil
-    ) throws -> Bool {
+    public func createSignedData(
+        data: Data,
+        keyHandle: KeyHandle,
+        trifleCertificates: Array<TrifleCertificate>
+    ) throws -> SignedData {
         
-        // TODO: validate PK in MobileCertificateRequest against certificate
+        // convert from TrifleCertificate to Certificate
+        let certificates = trifleCertificates.map({ trifleCert in
```

Comment:
do we need this mapping if we have an equivalent `verify` function that operates on the TrifleCertificate level?

Reply:
true, we can operate on the TrifleCertificate.verify level instead.